examples/connext_dds/CMakeLists.txt

```diff
@@ -50,248 +53,94 @@ if(STATIC_ANALYSIS)
-    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/builtin_qos_profiles/c++")
-    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/builtin_qos_profiles/c++03")
+# These conditionals could be replaced once we have updated to CMake 3.18.
+# To do this, we could replace the if inside the add_examples function
+# for cmake_language(EVAL CODE ...) and pass the condition as a string
+# parameter instead of passing a boolean.
+# Take a look of cmake_language in the documentation.
+
+# custom_transport requires D_BUILD_SHARED_LIBS=ON and cannot be compiled on
+# Windows because it uses Unix system calls to perform file operations such as
+# locking, reading and writing.
+if(WIN32 OR NOT BUILD_SHARED_LIBS)
+    set(SKIP_CUSTOM_EXTRA_ARGS
+     SKIP
+     SKIP_MESSAGE "it is not compatible with Windows and requires DBUILD_SHARED_LIBS=ON"
+    )
 endif()
 
-# builtin_topics
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/builtin_topics/c")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/builtin_topics/c++")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/builtin_topics/c++03")
-
-# coherent_presentation
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/coherent_presentation/c")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/coherent_presentation/c++")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/coherent_presentation/c++03")
-
-# content_filtered_topic
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/content_filtered_topic/c")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/content_filtered_topic/c++")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/content_filtered_topic/c++03")
-
-# content_filtered_topic_string_filter
-add_subdirectory(
-    "${CMAKE_CURRENT_SOURCE_DIR}/content_filtered_topic_string_filter/c")
-add_subdirectory(
-    "${CMAKE_CURRENT_SOURCE_DIR}/content_filtered_topic_string_filter/c++")
-add_subdirectory(
-    "${CMAKE_CURRENT_SOURCE_DIR}/content_filtered_topic_string_filter/c++03")
-
-# custom_content_filter
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/custom_content_filter/c")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/custom_content_filter/c++")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/custom_content_filter/c++03")
-
-# custom_flow_controller
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/custom_flow_controller/c")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/custom_flow_controller/c++")
-add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/custom_flow_controller/c++03")
+connextdds_add_examples_subdirectories(
+    EXAMPLES
+        "custom_transport"
+    ${SKIP_CUSTOM_EXTRA_ARGS}
+)
 
-# custom_transport
-if(NOT WIN32 AND BUILD_SHARED_LIBS)
-    add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/custom_transport/c")
-else()
-    message(
-        STATUS "Skipping custom_transport example: not compabile with Windows"
+# builtin_qos_profiles needs DBUILD_SHARED_LIBS=ON because it will require
+# to be linked against RTI's dynamic libraries so the monitoring library can
```

Comment:
```suggestion
# builtin_qos_profiles has to be built as a shared library because it is linked
# against RTI's dynamic libraries so the monitoring library can
```